### PR TITLE
Rearchitecting how the syntax highlighter library is unit tested

### PR DESCRIPTION
**Pull request by erunion**
## 🧰 What's being changed?

This completely rearchitects how `@readme/syntax-highlighter` is unit tested by creating a new drop-in test system that makes adding support for new languages TDD-driven. Now when adding a new language, you can create a new fixture, with a code sample, in `__tests__/__fixtures__/newLanguage` and the unit testing architecture will automatically detect it and run test assertions against it.

For details how it works, check the new contributing section in the README.

I've also added support for Objective-C++ and XHTML, as well as the following new language aliases:

* `aspx` for ASP.NET (and renamed ASP to ASP.NET in the process since that's its proper name)
* `m` for Objective-C
* `ps1` for PowerShell
* `styl` for Stylus

Finally, I've bulked up the README a bit to document all of these languages, and mode aliases, that we support. 🚀 

* [x] **🆕 I'm adding something new!**
* [ ] **🐛 I'm fixing a bug!**
* [ ] **📸 I've made some changes to the UI!**

### CODE REVIEW

File: packages/syntax-highlighter/__tests__/__fixtures__/dockerfile/sample.dockerfile [hunk 1]
```diff
@@ -0,0 +1 @@
+FROM alping:3.4
```
Comment (gratcliff) [inline packages/syntax-highlighter/__tests__/__fixtures__/dockerfile/sample.dockerfile:1]: Should this be `alpine`?

Comment (erunion) [reply packages/syntax-highlighter/__tests__/__fixtures__/dockerfile/sample.dockerfile:1]: 🥇

File: packages/syntax-highlighter/__tests__/__fixtures__/objc/sample.m [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ Build on OS X:
+ clang -framework Foundation -fobjc-arc objc.m -o objc
+
+ Build on Linux with GNUstep:
+ clang \`gnustep-config --objc-flags\` \`gnustep-config --base-libs\` -fobjc-nonfragile-abi -fobjc-arc objc.m -o objc
+ */
+
+#import <Foundation/Foundation.h>
+
+int main(void)
```
Comment (gratcliff) [inline packages/syntax-highlighter/__tests__/__fixtures__/objc/sample.m:11]: Do you need to pass in a `void` arg here? I think if you're trying to say it has no return value, the syntax is a bit different.

File: packages/syntax-highlighter/__tests__/index.test.js [hunk 3]
```diff
@@ -1,5 +1,12 @@
 const { mount, shallow } = require('enzyme');
+const path = require('path');
+const glob = require('glob');
+const fs = require('fs').promises;
+
 const syntaxHighlighter = require('..');
+const uppercase = require('../uppercase');
+
+const fixtures = glob.sync(path.join(__dirname, '/__fixtures__/*'));
```
Comment (gratcliff) [inline packages/syntax-highlighter/__tests__/index.test.js:9]: Lovely!!

File: packages/syntax-highlighter/__tests__/__fixtures__/objc/sample.m (hunk 2 above)
Comment (erunion) [reply packages/syntax-highlighter/__tests__/__fixtures__/objc/sample.m:11]: Pulled this example from https://github.com/leachim6/hello-world/blob/master/o/objective-c.m, which reminds me I need to pull in their license.

**Pull request merged**